dask_sql/mappings.py

```diff
@@ -191,7 +212,11 @@ def sql_to_python_value(sql_type: "SqlTypeName", literal_value: Any) -> Any:
         or sql_type == SqlTypeName.DATE
     ):
         if isinstance(literal_value, str):
-            literal_value = np.datetime64(literal_value)
+            try:
+                literal_value = np.datetime64(literal_value)
+            except ValueError:
+                literal_value = parse_datetime(literal_value)
+                literal_value = np.datetime64(literal_value)
```

Comment:
Wonder if at this point it would just make sense to always parse the datetime? Or does this not work in cases where `np.datetime64(literal_value)` wouldn't initially fail?